src/Constraint/ConstraintConfig.php

```diff
@@ -129,4 +134,23 @@ public function setConstructorCallback(callable $callback): self
 
         return $this;
     }
+
+    public function getValueProvider(): ?callable
+    {
+        return $this->valueProvider;
+    }
+
+    /**
+     * Callback function to allow for a custom value provider. For instance for final classes. The argument
+     * is the typehint of the value, the return value should be the provided value. Return <code>null</code>
+     * to skip the value provider and use the default value providers.
+     *
+     * @param callable(Type): mixed $valueProvider
```

Comment:
I think a better dex experience would be to have `callable(class-string)`, instead of the PHPDocumentor Type class.
The PHPDocumentor lib is an internal dependency, and this makes it impossible to replace it with another library.

Reply:
Ah right. agree. Fixed